File: php/スキルチェック過去問題セット/野球の審判.php
<?php
    // 自分の得意な言語で
    // Let's チャレンジ！！

    $count = trim(fgets(STDIN)); // 入力値（カウント）
    
    for($i = 0; $i < $count; $i++){
        
        // 入力値（ストライク、ボール）を配列に格納
        $array[$i] = trim(fgets(STDIN)); 
        
        // 特定の配列値をカウント
        $count_array = array_count_values($array);
        
        // 配列にストライクがあった場合
        if(!empty($count_array['strike'])){
            
            // ストライクが３つの場合
            if($count_array['strike'] == 3){
                echo "out!";
                break;
            }
        
        }
        
        // 配列にボールがあった場合
        if(!empty($count_array['ball'])){
            // ボールが４つの場合
            if($count_array['ball'] == 4){
                echo "fourball!";
                break;
            }
        }
        
        // 入力値の出力
        echo $array[$i] . "!" . "\n";
    }
    
?>

File: python/スキルチェック過去問題セット/野球の審判.py
# coding: utf-8
# 自分の得意な言語で
# Let's チャレンジ！！
count = int(input())

strike_count = 0
ball_count = 0

for i in range(count):
    str = input()
    if str == 'strike':
        if strike_count == 2:
            print('out!')
            break
        else:
            print('strike!')
        strike_count = strike_count + 1
        
    elif str == 'ball':
        if ball_count == 3:
            print('fourball!')
            break
        else:
            print('ball!')
        
        ball_count = ball_count + 1
    else:
        pass
